packages/block-editor/src/components/list-view/block-contents.js

```diff
@@ -61,23 +65,96 @@ const ListViewBlockContents = forwardRef(
 			? selectedClientIds
 			: [ clientId ];
 
+		const { blockName, blockAttributes } = useSelect(
+			( select ) => {
+				const blockObject =
+					select( blockEditorStore ).getBlock( clientId );
+				return {
+					blockName: blockObject?.name,
+					blockAttributes: blockObject?.attributes,
+				};
+			},
+			[ clientId ]
+		);
+
+		const { updateBlockAttributes } = useDispatch( blockEditorStore );
+
+		const metaDataSupport = getBlockSupport(
+			blockName,
+			'__experimentalMetadata',
+			false
+		);
+
+		const supportsBlockNaming = !! (
+			true === metaDataSupport || metaDataSupport?.name
+		);
+
+		const { isLocked } = useBlockLock( clientId );
```

Comment:
Since block select button uses block lock internally maybe block inout should as well?

Reply:
> maybe block inout should as well?

I didn't follow this sorry. Can you explain?